study/Cloud Computing/k8s-files/keycloak/Dockerfile

```diff
@@ -0,0 +1,34 @@
+# Use the official Keycloak image as the base
+FROM quay.io/keycloak/keycloak:latest as builder
+
+# Enable health and metrics
+ENV KC_HEALTH_ENABLED=true
+ENV KC_METRICS_ENABLED=true
+
+# Configure database vendor
+ENV KC_DB=postgres
+
+WORKDIR /opt/keycloak
+
+# Generate a demo key store (not for production)
+RUN keytool -genkeypair -storepass password -storetype PKCS12 -keyalg RSA -keysize 2048 \
+    -dname "CN=server" -alias server \
+    -ext "SAN:c=DNS:localhost,IP:127.0.0.1" -keystore conf/server.keystore
+
+# Optimize the Keycloak build
+RUN /opt/keycloak/bin/kc.sh build
+
+FROM quay.io/keycloak/keycloak:latest
+COPY --from=builder /opt/keycloak/ /opt/keycloak/
+
+# Set environment variables for PostgreSQL and Keycloak
+ENV KC_DB=postgres
+ENV KC_DB_URL=jdbc:postgresql://localhost:5432/keycloak
+ENV KC_DB_USERNAME=keycloak
+ENV KC_DB_PASSWORD=mypassword
+ENV KC_BOOTSTRAP_ADMIN_USERNAME=admin
+ENV KC_BOOTSTRAP_ADMIN_PASSWORD=admin
+ENV KC_HOSTNAME=http://localhost:32080
```

Comment:
This is a big pain point, the problem with that is that making KC_HOSTNAME localhost then that means that every redirect inside the keycloak will always point to the localhost which in the case of opening it from mobile devices will lead to errors, scenario:

- I start the kubernetes with make up
- I can acces keycloak through localhost of  the host machine (windows machine)
- The mobile phone in the same local area network enters the ip of the windows machine 10.151.130.198
-The first keycloak screen appears
- By entering the admin user and password the keycloak then trys to show the next screen (the dashboard) but then an error occurs on the mobile device because the url now has localhost inside which is wrong because the mobile device is not the localhost, so the host should be the one of the windows machine.


We still have to discuss this to fix the error and have a lab that is functional.

Reply:
added your local host ip instead